ooni/tests/mocks.py

```diff
@@ -242,3 +242,19 @@ def isSupported(self):
 
     def isReachable(self):
         return defer.succeed(True)
+
+
+class MockDnsResponsePayload(object):
+    def dottedQuad(self):
+        return '127.0.0.1'
+
+
+class MockDnsResponse(object):
+    payload = MockDnsResponsePayload()
+
+
+def mock_dns_lookup_address():
+    response = (
+        [MockDnsResponse()],
+    )
+    yield response
```

Comment:
you must not yield here, but instead return an already fire deferred via `return defer.succeed(response)`